### PR TITLE
Add option for SVHN to get train and extra as one dataset

### PR DESCRIPTION
**Pull request by xksteven**
All state of the art networks train on both train and extra data together. I added this option.

### CODE REVIEW

**Comment by fmassa** [comment]
Thanks for the PR.
While I think that the PR is ok as is, I'd like to point out that this functionality could be implemented by using a [`concatdataset` from tnt](https://github.com/pytorch/tnt/blob/master/torchnet/dataset/concatdataset.py). I think it's generic enough so that it could be merged into master. 
Let me check what the others think.

In any case, I noticed that you also subtracted one from the labels (making it zero-based), and this change should be kept.

**Comment by xksteven** [comment]
The failing checks are related to lint which has some error about flake8 not related to my pull request.  

Thanks for considering the pull request and informing me about the concatdataset option.  I still think the pull request is relevant because most do not see the "extra" dataset from svhn as an independent dataset but instead as something that should be optionally included with the train dataset.  As such it would be nice to have it as option from the start.

I ran into errors in pytorch with 1 based indexing for different loss functions so I added in this fix as well.  Let me know if you think there are any changes that I should make before the pull request is accepted.
